### PR TITLE
tag expressions: add go implementation

### PR DESCRIPTION
**Pull request by charlierudolph**
## Motivation and Context

I'm working on the shared binary and need this since I'm programming in go.

## How Has This Been Tested?

Ported over the tests (used javascript as the template)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Notes:

I was able to test through docker on my machine but we need to push a docker image for this to work on CI.

### CODE REVIEW

**Comment by aslakhellesoy** [comment]
I'm getting a build error:

```
cd tag-expressions/go
make

go test
# _/Users/aslakhellesoy/git/cucumber/cucumber/tag-expressions/go
parser_test.go:6:2: cannot find package "github.com/cucumber/cucumber/tag-expressions/go" in any of:
	/usr/local/Cellar/go/1.9.2/libexec/src/github.com/cucumber/cucumber/tag-expressions/go (from $GOROOT)
	/Users/aslakhellesoy/git/cucumber/cucumber/tag-expressions/go/src/github.com/cucumber/cucumber/tag-expressions/go (from $GOPATH)
FAIL	_/Users/aslakhellesoy/git/cucumber/cucumber/tag-expressions/go [setup failed]
make: *** [default] Error 1
```

**Comment by charlierudolph** [comment]
@aslakhellesoy go requires the src to live in a specific directory. As you can see from my changes: `/go/src/github.com/cucumber/cucumber`. Can we make that the workdir and always mount the whole project?  I'll create a PR with those changes

**Comment by aslakhellesoy** [comment]
It's great to be able to run the build in Docker, but I don't think it should be a requirement. `gherkin/go` builds fine both inside and outside Docker, and the source doesn't have to be in a particular directory.

Can't we make that work for tag-expressions too?

File: tag-expressions/go/parser_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
+package tagexpressions_test
+
+import (
+	"testing"
+
+	tagexpressions "."
```
Comment (charlierudolph) [inline tag-expressions/go/parser_test.go]: This is what was done in gherkin/go. This is never something I have done and think is too much against the go way of doing things. I would prefer we setup go to have its own docker image

Comment (aslakhellesoy) [reply tag-expressions/go/parser_test.go]: I think it's more pragmatic to add a dot in a test than maintaining several Docker images.

Comment (aslakhellesoy) [reply tag-expressions/go/parser_test.go]: It should also be possible to build the code without having to use Docker.

**Comment by charlierudolph** [comment]
Well this is now ready. The build failure is from CircleCI which I am still working on. Travis is passing.

**Comment by aslakhellesoy** [comment]
Great! I'll go ahead and merge in the next few hours. Very exciting to see this Go stuff advancing!

**Pull request merged**